workspaces/services/guacamole-azure-dsvm/terraform/dsvm.tf

```diff
@@ -0,0 +1,128 @@
+  resource_group_name = data.azurerm_resource_group.ws.name
+}
+
+resource "azurerm_network_security_rule" "allow-rdp-inbound-from-vnet-dsvm" {
+  name                        = "RDP"
+  priority                    = 1010
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = 3389
+  source_address_prefix       = "VirtualNetwork"
+  destination_address_prefix  = "*"
+  network_security_group_name = azurerm_network_security_group.dsvmnsg.name
+  resource_group_name         = data.azurerm_resource_group.ws.name
+}
+
+resource "azurerm_network_security_rule" "deny-all-inbound-override-dsvm" {
+  access                      = "Deny"
+  destination_address_prefix  = "*"
+  destination_port_range      = "*"
+  direction                   = "Inbound"
+  name                        = "deny-inbound-override"
+  network_security_group_name = azurerm_network_security_group.dsvmnsg.name
+  priority                    = 900
+  protocol                    = "*"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_address_prefix       = "*"
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_security_rule" "deny-outbound-override-dsvm" {
+  access                      = "Deny"
+  destination_address_prefix  = "*"
+  destination_port_range      = "*"
+  direction                   = "Outbound"
+  name                        = "deny-outbound-override"
+  network_security_group_name = azurerm_network_security_group.dsvmnsg.name
+  priority                    = 4096
+  protocol                    = "*"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_address_prefix       = "*"
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_interface_security_group_association" "dev_vm_nsg_association" {
+  network_interface_id      = azurerm_network_interface.internal.id
+  network_security_group_id = azurerm_network_security_group.dsvmnsg.id
+}
+
+resource "random_string" "username" {
+  length      = 4
+  upper       = true
+  lower       = true
+  number      = true
+  min_numeric = 1
+  min_lower   = 1
+  special     = false
+}
+
+resource "random_password" "password" {
+  length           = 16
+  lower            = true
+  min_lower        = 1
+  upper            = true
+  min_upper        = 1
+  number           = true
+  min_numeric      = 1
+  special          = true
+  min_special      = 1
+  override_special = "_%@"
+}
+
+resource "azurerm_virtual_machine" "dsvm" {
+  name                             = local.vm_name
+  location                         = data.azurerm_resource_group.ws.location
+  resource_group_name              = data.azurerm_resource_group.ws.name
+  network_interface_ids            = [azurerm_network_interface.internal.id]
+  vm_size                          = "Standard_DS1_v2"
```

Comment:
Seems very small machine for the use-case.

Reply:
one e2e will work we will upgrade. no reason to do it now